server/src/main/java/org/apache/druid/metadata/SqlSegmentsMetadataQuery.java

```diff
@@ -619,6 +639,13 @@ private Query<Map<String, Object>> buildSegmentsTableQuery(
       appendConditionForIntervalsAndMatchMode(sb, intervals, matchMode, connector);
     }
 
+    if (!CollectionUtils.isNullOrEmpty(versions)) {
+      final String versionsStr = versions.stream()
+                                          .map(version -> "'" + version + "'")
+                                          .collect(Collectors.joining(","));
+      sb.append(StringUtils.format(" AND version IN (%s)", versionsStr));
```

Comment:
should we enforce some limit on the number of versions here? I think in practice most users wouldnt be specifying too many versions at once.

Reply:
Good point. However, we don't enforce an upper bound on some related parameters like `limit` and `batchSize`, so I'm unsure if we want to do it for `versions` size. Also, the maximum size for `IN` would also depend on the underlying metadata store itself, so my suggestion would be to roll this out without any size restriction right out of the bat, and revisit this later if needed